cfgov/unprocessed/js/modules/VideoPlayer.js

```diff
@@ -0,0 +1,206 @@
+
+'use strict';
+
+var _assign = require( './util/assign' ).assign;
+var _noopFunct = require( './util/standard-type' ).noopFunct;
+var DOM_INVALID = require( '../config/error-messages-config' ).DOM.INVALID;
+var jsLoader = require( './util/js-loader' );
+
+var CLASSES = Object.freeze( {
+    VIDEO_PLAYER_SELECTOR:     '.video-player',
+    VIDEO_PLAYING_STATE:       'video-playing',
+    IFRAME_CLASS_NAME:         'video-player_iframe',
+    IFRAME_CONTAINER_SELECTOR: '.video-player_iframe-container',
+    PLAY_BTN_SELECTOR:         '.video-player_play-btn',
+    CLOSE_BTN_SELECTOR:        '.video-player_close-btn'
+} );
+
+function VideoPlayer( element, options ) {
+  _this = this;
+  options = options || {};
+  this.baseElement = _ensureElement( element, options.createIFrame );
+
+  // TODO: Add utility function for dataset. Oh dear IE, how I love you.
+  this.iFrameProperties = _assign( this.baseElement.dataset || {}, this.iFrameProperties );
+  _setChildElements( this.childElements );
+  _initEvents();
+  this.init.apply( this, arguments );
+}
+
+VideoPlayer.extend = function extend( attributes ) {
+  function child() {
+    this._super = VideoPlayer.prototype;
+    return VideoPlayer.apply( this, arguments );
+  }
+  child.prototype = Object.create( VideoPlayer.prototype );
+  _assign( child.prototype, attributes );
+  child.init = VideoPlayer.init;
+
+  return child;
+};
+
+VideoPlayer.init = function init( selector ) {
+  selector = selector || '[class*="video-player"]:not([class*="video-player__"])';
+  // There should only be one video player on the page.
+
+  var videoPlayerElement = document.querySelector( selector );
+  var videoPlayer;
+  if ( videoPlayerElement ) {
+    videoPlayer = new this( videoPlayerElement );
+  }
+
+  return videoPlayer;
+};
+
+// PRIVATE functions and properties.
+var _isIframeLoaded;
+var _isIframeLoading;
+var _isVideoPlaying
+var _isVideoStopped
+var _this = this;
+
+function _createElement( tagName, properties ) {
+    var createdElement = document.createElement( tagName );
+    var property;
+    var hasOwnProperty = Object.hasOwnProperty.bind( properties );
+
+    for ( property in properties || {} ) {
+      if ( hasOwnProperty( property ) ) {
+        createdElement.setAttribute( property, properties[property] );
+      }
+    }
+
+    return createdElement;
+}
+
+function _attachIFrame() {
+    var iFrameElement = _this.childElements.iFrame;
+    var iFrameContainerElement = _this.childElements.iFrameContainer;
+    var state = _this.state;
+
+    if ( state.isIframeLoaded === false && state.isIframeLoading === false &&
+         iFrameElement === null ) {
+      iFrameElement = _createElement( 'iframe', _this.iFrameProperties );
+      iFrameElement.classList.add(  CLASSES.IFRAME_CLASS_NAME );
+      iFrameElement.setAttribute('frameborder', 0 );
+      state.isIframeLoading = true;
+      iFrameElement.onload = function iframeLoad() { state.isIframeLoaded = true; };
```

Comment:
So I left this as-is because the element is getting created directly above and the sub classes don't have `onLoad` properties (Although they could).